stdlib/public/core/StringLegacy.swift

```diff
@@ -33,14 +33,25 @@ extension String {
       self = count == 0 ? "" : repeatedValue
       return
     }
+    guard !repeatedValue.isEmpty else {
+        self = ""
+        return
+    }
 
-    // TODO(String performance): We can directly call appendInPlace
-    var result = String()
-    result.reserveCapacity(repeatedValue._guts.count &* count)
-    for _ in 0..<count {
-      result += repeatedValue
+    var repeatedValue = repeatedValue
+    self = repeatedValue.withUTF8 { repeatedUTF8 in
+      String(_uninitializedCapacity: repeatedUTF8.count * count) { buffer in
+        var total = 0
+        for i in 0..<count {
+          let offset = i * repeatedUTF8.count
+          let range = offset ..< offset + repeatedUTF8.count
+            _ = UnsafeMutableBufferPointer(rebasing: buffer[range])
```

Comment:
Nit: Indentation.

Reply:
> Question: Is there no other way to initialize a chunk of the buffer than to create another mutable buffer pointer and then throw it away?

`UnsafePointer<T>` has `advanced(by:)`, which anyway produces `UnsafePointer<T>` - new pointer.
(Interesting why [docs for `UnsafePointer`](https://developer.apple.com/documentation/swift/unsafepointer) misses this method).

I don't see anything like `advance()`.

`UnsafeMutableBufferPointer<T>` don't have `advanced(by:)` which might look like a gap in API.

But anyway, it would have produced a new pointer.

The answer for your question: I don't know a way to make it more effective and doubt if such a way exists.

Reply:
> Nit: Indentation.

fixed

Reply:
What about using the raw pointer or raw buffer pointer APIs?

Reply:
Your code and mine both create range and pointer inside loop. Generated code should be more or less the same. The only difference is `copyMemory(from:)` vs `initialise(from:)`.
if `copyMemory(from:)` is just a wrapper around `memcpy` it should be more effective. The more effective the bigger `repeatedValue` is as I expect `memcpy` to be be vectorised in C library.

Reply:
Right, and it gets rid of an addition operation in the tight loop since there's no need to calculate the upper bound :)

I also don't know if discarding the return value as you have it here is going to impact the generated code, but perhaps it's a little more elegant not to have to even if it makes no difference in the compiled binary.

(By the way, there's also no need to add to `total` step by step, as I don't think the copying can fail, so you can return the final value at the end.)